### PR TITLE
bug : mongobi issue with parsing schema file with nested fields (Scaffoldin…

### LINKED ISSUE #55: mongobi issue with parsing schema file with nested fields (Scaffolding bug)
Consider MongoDB Document 
`{
    "_id" : ObjectId("5c8a215c46e0fb0001055061"),
    "details" : {
        "name" : "asdfsadf",
        "status" : "asdfdsf",
        "severity" : "asdfsadf"
    },
    "projectId" : "5c6eb78046e0fb0001ee0b32",
.
.
.
}`

Schema file generate by cubejs contains measure names with "." for nested fields as below

`cube(`Flow`, {
  sql: `SELECT * FROM 1224.flow`,
  
  joins: {
    
  },
  
  measures: {
    count: {
      type: `count`,
      drillMembers: [createdbyuser, **details.name**, projectid, createddate, modifieddate]
    }
  },
  
  dimensions: {
    **details.owner**: {
      sql: `details.owner`,
      type: `string`
    },
    
    createdbyuser: {
      sql: `createdByUser`,
      type: `string`
    },
    
    **details.description**: {
      sql: `details.description`,
      type: `string`
    },
    
    **details.name**: {
      sql: `details.name`,
      type: `string`
    },
    
    **details.severity**: {
      sql: `details.severity`,
      type: `string`
    },
.
.
.
`

This throws error while parsing

Recompiling schema: {"version":"default_schema_version_bc194c45a5c745c8a93e4b0515c9dcc4"}
Internal Server Error: {"error":"Error: Compile errors:\nSyntaxError: Unexpected token, expected , (16:11)\n    at Parser.pp$5.raise (/home/kru/antphant/git/poc/node_modules/babylon/lib/index.js:4454:13)\n    at Parser.pp.unexpected (/home/kru/antphant/git/poc/node_modules/babylon/lib/index.js:1761:8)\n    at Parser.pp.expect (/home/kru/antphant/git/poc/node_modules/babylon/lib/index.js:1749:33)\n    at Parser.pp$3.parseObj (/home/kru/antphant/git/poc/node_modules/babylon/lib/index.js:3978:12)\n    at Parser.pp$3.parseExprAtom (/home/kru/antphant/git/poc/node_modules/babylon/lib/index.js:3719:19)\n    at Parser.pp$3.parseExprSubscripts (/home/kru/antphant/git/poc/node_modules/babylon/lib/index.js:3494:19)\n    at Parser.pp$3.parseMaybeUnary (/home/kru/antphant/git/poc/node_modules/babylon/lib/index.js:3474:19)\n    at Parser.pp$3.parseExprOps (/home/kru/antphant/git/poc/node_modules/babylon/lib/index.js:3404:19)\n    at Parser.pp$3.parseMaybeConditional (/home/kru/antphant/git/poc/node_modules/babylon/lib/index.js:3381:19)\n    at Parser.pp$3.parseMaybeAssign (/home/kru/antphant/git/poc/node_modules/babylon/lib/index.js:3344:19)\n    at ErrorReporter.throwIfAny (/home/kru/antphant/git/poc/node_modules/@cubejs-backend/schema-compiler/compiler/DataSchemaCompiler.js:42:13)\n    at DataSchemaCompiler.throwIfAnyErrors (/home/kru/antphant/git/poc/node_modules/@cubejs-backend/schema-compiler/compiler/DataSchemaCompiler.js:149:23)\n    at repository.dataSchemaFiles.then.then (/home/kru/antphant/git/poc/node_modules/@cubejs-backend/schema-compiler/compiler/DataSchemaCompiler.js:99:14)","authInfo":{"iat":1553003564,"exp":1553089964}}

### PR DESCRIPTION
**Pull request by manojown**
resolve #55 

**Occurence** 
 - Its occurred when we parse nested field's schema. 

**FIX**
  - Ast parser (babylon parser) failed to parse ```any.any``` key syntax so replace ```.``` with ```_``` in ScaffoldingTemplate.

### CODE REVIEW

File: packages/cubejs-schema-compiler/scaffolding/ScaffoldingTemplate.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ class ScaffoldingTemplate {
   }
 
   memberName(member) {
-    return inflection.camelize(member.title.replace(/\s+/g, '_').toLowerCase(), true);
+    return inflection.camelize(member.title.replace(/\s|\.+/g, '_').toLowerCase(), true);
```
Comment (paveltiunov) [inline packages/cubejs-schema-compiler/scaffolding/ScaffoldingTemplate.js]: I believe it should be `/\s+|\./g`. In this multiple space rule kept intact.

Comment (manojown) [reply packages/cubejs-schema-compiler/scaffolding/ScaffoldingTemplate.js]: ya right . Thanks

**Comment by paveltiunov** [comment]
@manojown Looks great! Did you have a chance to test it against Mongo BI connector with nested fields?

**Comment by manojown** [comment]
@paveltiunov yes i did and its seems like all things work perfectly accept nested field issue while filter .
**Issue** 
Screen Shot 2019-03-31 at 10 27 32 PM

**REASON**
I thinks its occurred because of bad formats like 

**Failed scenario** 
I tried in sql client and its failed because of
Screen Shot 2019-03-31 at 10 41 07 PM
 sql client throw error - 
 
Screen Shot 2019-03-31 at 10 43 49 PM

**Work perfectly with**
Screen Shot 2019-03-31 at 10 40 09 PM

Its work when i wrap nested call in backtick .

I think we should create different issue for this bug  right ?

**Comment by paveltiunov** [comment]
@manojown Hey Manoj! Yep. Seems it's another one issue. Looks like nested fields should be escaped. This one I think is good to merge

**Pull request merged**